lib/Storage/MongoDbTimeSeriesStore.php

```diff
@@ -161,10 +163,12 @@ public function addStat($statData)
                     $this->mediaItems[$statData['mediaId']] = $media;
 
                 } catch (NotFoundException $error) {
-                    // Media not found ignore and log the stat
-                    $this->log->error('Media not found. Media Id: '. $statData['mediaId'] .',Layout Id: '. $statData['layoutId']
-                        .', FromDT: '.$statData['start'].', ToDt: '.$statData['end'].', Type: '.$statData['type']
-                        .', Duration: '.$statData['duration'] .', Count '.$statData['count']);
+
```

Comment:
This is good - but can we go one step further do you think?

If we have `$this->mediaItemsNotFound`, we could not do the look up next time and save ourselves lots of processing? This also makes me wonder if we should cache the result as well - if we have 100 stats and they are all the same mediaId, we make this call 100 times and it could be 1 time?